lib/rspec/mocks/message_expectation.rb

```diff
@@ -683,6 +725,24 @@ def call(*_args_to_ignore, &_block)
       end
     end
 
+    # Handles the implementation of an `and_invoke` implementation.
+    # @private
+    class AndInvokeImplementation
+      def initialize(procs_to_invoke)
+        @procs_to_invoke = procs_to_invoke
+      end
+
+      def call(*args, &block)
+        proc = if @procs_to_invoke.size > 1
```

Comment:
We could check for `@procs_to_invoke.one?`, it's [there since 1.8.7](https://ruby-doc.org/core-1.8.7/Enumerable.html#method-i-one-3F).